internal/resource-request-operator/utils.go

```diff
@@ -51,3 +52,29 @@ func (r *ResourceRequestReconciler) generateResourceOffer(request *discoveryv1al
 	klog.Infof("%s -> %s Offer: %s", r.ClusterID, op, offer.ObjectMeta.Name)
 	return nil
 }
+
+func (r *ResourceRequestReconciler) createForeignCluster(resourceRequest *discoveryv1alpha1.ResourceRequest) error {
+	foreignCluster := &discoveryv1alpha1.ForeignCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: resourceRequest.Spec.ClusterIdentity.ClusterID,
+			Labels: map[string]string{
+				discovery.DiscoveryTypeLabel: string(discovery.IncomingPeeringDiscovery),
+				discovery.ClusterIDLabel:     resourceRequest.Spec.ClusterIdentity.ClusterID,
+			},
+		},
+		Spec: discoveryv1alpha1.ForeignClusterSpec{
+			ClusterIdentity: resourceRequest.Spec.ClusterIdentity,
+			Namespace:       resourceRequest.Namespace,
+			Join:            false,
+			DiscoveryType:   discovery.IncomingPeeringDiscovery,
+			AuthURL:         resourceRequest.Spec.AuthURL,
+		},
+	}
+
+	err := r.Create(context.Background(), foreignCluster)
```

Comment:
It seems to me at this point the resourcerequest should become the owner of this foreigncluster to ensure eventual deletion. @aleoli Is that correct?

Reply:
No, it can't, it is a namespaced resource while the ForeignCluster is cluster-wide. The deletion logic is inside the ForeignCluster Operator.
It has to be set the opposite, the ResourceRequest ahs to be owned by the ForeignClsuter to ensure FC Status updates on incoming peering chages.

Reply:
Got it. Indeed, as discussed, it should be configured the reverse owner-reference (the resource request is owned by the foreigncluster), to ensure the status is correctly reconciled. Once those two operations are performed, it is probably better to return and create the resourceoffer in the next reconcile, to prevent self-created conflicts